### ISSUE TITLE
Back button navigation breaks

### ISSUE THREAD

**Issue by hackel** [ovity/octotree#727]
### Description
Octotree completely breaks navigation on Github.  Moving back does not bring you to the page you expect.  Start on the root of a repo, browse a few levels deep, then press back until you get to the start of your navigation history, and you don't end up at the root of the repo like you would expect.  From this point, navigating forward and back the page it decides to go to seems to be random, I can't even see a pattern to it.  Seems like perhaps both Github and this extension are messing with the browser history API and conflicting with each other.

### Environment

* Octotree version: 3.0.7
* Browser & version: Firefox 68b3

**Comment by nphi** [comment]
Thanks for reporting the issue. I will investigate it and come back to you later.

**Comment by nphi** [comment]
Hi @hackel ,

Could you give us a screenshot so we could understand more your issue ? I have tested it and the navigation seems work fine to me. 

Looking forward hearing from you.

**Comment by hackel** [comment]
There isn't really anything to screenshot other than Github itself.  It seem to happen all the time, but the following flow seems to be reproducing it reliably.  With Octotree running on your repo, I do the following (in GitHub, not touching Octotree):

1. click on dist > to to dist
2. press back > go to root
3. press forward > go to dist
4. press back > stay on dist (or it gets reloaded?), with no further back navigation possible
5. press forward > go to root (what?)
6. press back > go to dist
7. press forward > stay on dist 
8. press back > go to root

This is just one example.  I had a hard time finding something that would work reliably.  In my daily use, I find it comes up frequently when I need to go back to the root of a repo, but pressing back stays on the same page with no further back navigation possible, so I have to click on a link back to the repo root.

**Comment by nphi** [comment]
Thanks for your cooperation. I will investigate the issue ASAP.

**Comment by edemaine** [comment]
FWIW, I have had this behavior for months and just tracked it down to Octotree. Disabling Octotree fixes the issue.

**Comment by vegeta897** [comment]
I've been having this issue intermittently ever since installing this extension more than a year ago. I disabled it today and I can no longer make it happen.

It seems to happen mostly when quickly navigating back after clicking a file/dir in a repo. The URL updates to the previous one, but the page continues to display what it did before going back.

I re-enabled the extension when I found this issue to make sure I could repro the issue. At first I could not, but then I tried hovering over the OctoTree tab on the left to make it open. After that, I could get it to happen.

So my best repro steps are:

1. Open a repo page
2. Hover over the OctoTree tab on the left to make the panel appear -- then let it close
3. Navigate into a directory and then to a subdirectory or file, and quickly use the Back button in your browser
4. Eventually, the page displayed will become out of sync with the URL in the browser
